DOC/RLS: update release notes for 0.5 (#202)
geopandas/pyogrio: CHANGES.md

-   Support for appending to an existing data source when supported by GDAL by
    passing `append=True` to `pyogrio.write_dataframe` (#197).

### Improvements

-   It is now possible to pass GDAL's dataset creation options in addition
    to layer creation options in `pyogrio.write_dataframe` (#189).
-   When specifying a subset of `columns` to read, unnecessary IO or parsing
    is now avoided (#195).

> -   In floating point columns, NaN values are now by default written as "null"

Reviewer: Should we list this as a potentially breaking change?

Author: As a separate section? Or just mention that is is potentially breaking in this list item? 
(I am just not sure how much this will affect people in practice, although I just checked and for example with opening a shapefile in QGIS, is does differ, i.e. it now sees it as missing value instead of as NaN)

Reviewer: As a separate section, as a sibling of enhancements:

```
### Potentially breaking changes

-  In floating point columns, NaN values are now...

```

    instead of NaN, but with an option to control this (pass `nan_as_null=False`
    to keep the previous behaviour) (#190).

### Packaging

-   The GDAL library included in the wheels is updated from 3.4 to GDAL 3.6.2,

>     and is now built with GEOS and with sqlite with rtree support enabled

Reviewer: ```suggestion
    and is now built with GEOS and sqlite with rtree support enabled
```

>     (e.g. allowing to write a spatial index for GeoPackage).

Reviewer: ```suggestion
    (which allows writing a spatial index for GeoPackage).
```
